plugins/providers/gcs/resource.go

```diff
@@ -0,0 +1,28 @@
+package gcs
+
+import "github.com/odpf/guardian/domain"
+
+const (
+	ResourceTypeBucket = "bucket"
+)
+
+type Bucket struct {
+	Name string
+}
+
+func (b *Bucket) fromDomain(r *domain.Resource) error {
+	if r.Type != ResourceTypeBucket {
+		return ErrInvalidResourceType
+	}
+
+	b.Name = r.URN
+	return nil
+}
+
+func (b *Bucket) toDomain() *domain.Resource {
+	return &domain.Resource{
+		Type: ResourceTypeBucket,
+		URN:  b.Name,
```

Comment:
does URN always equal to the name (or vice versa)?

Reply:
yes, the resource URN(in guardian) is the actual resource name(in the provider) which the user might want to create an appeal for. Might differ from provider to provider. For a `bucket` resource, @rahmatrhd and I discussed having URN the same as the bucket name. And for the `object`(no longer supported) it was the complete path. 

Reference: the Resource Config for Bucket and Object https://github.com/odpf/guardian/issues/194